### ISSUE TITLE
Configuration design

### ISSUE THREAD

**Issue by humboldtux** [humboldtux/cisgo#4]
This is an issue to discuss the design of the configuration.

We need a main configuration file, let's say `cisgo.conf`, where to configure general parameters (logging, admin email, timeout).

We need a directory, let's say `.cisgo.d`, which will contain the configuration of the devices, one file per device.

We need to define the syntax of  these files.

Do we use a custom parser, à la [Inspeqtor](https://github.com/mperham/inspeqtor/tree/master/conf), which shares the same hierarchy?

Do we use a third party package? Which one? [hcl](https://github.com/hashicorp/hcl)?

**Comment by gernest** [comment]
@humboldtux we can just use `json` for this, no need for third party libraries. I have written a lot of configuration implementation to say that that json is enough.

**Comment by humboldtux** [comment]
Ok for JSON: a lot of people don't like to use it as a configuration syntax for readibility issues, but our config syntax won't be complex, so JSON is a good fit.

**Comment by gernest** [comment]
@humboldtux yeah! so we are good with `json` right? Now later I will write some tests for loading the configuration and I will walk you through the implementation ( You will be writing the code ) or do you have any better idea than this?

**Comment by humboldtux** [comment]
Yes, sounds perfect: you writing the tests, and me writing the code to make it pass.

**Comment by gernest** [comment]
:+1:

**Comment by gernest** [comment]
@humboldtux  I have added the test, its your turn to make it pass :smile: .
